### PR TITLE
[docs] Clarify withContiguousStorageIfAvailable documentation

### PR DESCRIPTION
**Pull request by glessard**
Improves documentation of the `withContiguousStorageIfAvailable` and `withContiguousMutableStorageIfAvailable` functions. This does not change the requirements, but states them more forcefully.

### CODE REVIEW

**Comment by glessard** [comment]
@swift-ci please smoke test

File: stdlib/public/core/MutableCollection.swift [hunk 1]
```diff
@@ -173,33 +173,34 @@ where SubSequence: MutableCollection
...
-  /// Call `body(p)`, where `p` is a pointer to the collection's
-  /// mutable contiguous storage.  If no such storage exists, it is
-  /// first created.  If the collection does not support an internal
-  /// representation in a form of mutable contiguous storage, `body` is not
-  /// called and `nil` is returned.
-  ///
-  /// Often, the optimizer can eliminate bounds- and uniqueness-checks
-  /// within an algorithm, but when that fails, invoking the
-  /// same algorithm on `body`\ 's argument lets you trade safety for
-  /// speed.
```
Comment (lorentey) [inline stdlib/public/core/MutableCollection.swift:185]: I think we should keep these documented, even if they are deprecated

File: stdlib/public/core/Sequence.swift [hunk 2]
```diff
@@ -384,18 +384,27 @@ public protocol Sequence {
...
+  /// within an algorithm. When that fails, however, invoking the same
+  /// algorithm on `body`\ 's argument may let you trade safety for speed.
+  ///
   /// A `Collection` that provides its own implementation of this method
-  /// must also guarantee that an equivalent buffer of its `SubSequence` 
-  /// can be generated by advancing the pointer by the distance to the
-  /// slice's `startIndex`.
+  /// must guarantee that contiguous storage to any valid `SubSequence`
+  /// can be generated by slicing `buffer` with a range formed from the
+  /// distances to the subsequence's `startIndex` and `endIndex`, respectively.
```
Comment (lorentey) [inline stdlib/public/core/Sequence.swift:400]: This passage is a definite improvement, but I find it a bit difficult to understand what it means. Is it equivalent to stating that the order of elements in the buffer must match the order they appear in the collection?

**Review by lorentey — APPROVED** [review]
Looks great! 👍 A couple of comments below

File: stdlib/public/core/Sequence.swift (hunk 2 above)
Comment (glessard) [reply stdlib/public/core/Sequence.swift:400]: I think it does. That would be a good way to say it.

File: stdlib/public/core/MutableCollection.swift (hunk 1 above)
Comment (glessard) [reply stdlib/public/core/MutableCollection.swift:185]: I restored the documentation, even though they're both deprecated and internal.

**Comment by glessard** [comment]
@swift-ci please smoke test

**Comment by glessard** [comment]
@swift-ci please smoke test

**Comment by glessard** [comment]
@swift-ci please smoke test and merge

**Comment by glessard** [comment]
@swift-ci please smoke test and merge

**Review by stephentyrone — APPROVED** [review]

**Review by lorentey — APPROVED** [review]

**Comment by glessard** [comment]
@swift-ci please test macOS platform

**Comment by glessard** [comment]
@swift-ci please smoke test linux platform

**Comment by glessard** [comment]
@swift-ci please test macOS platform

**Comment by glessard** [comment]
The Linux errors are in swift-docc tests.

**Comment by glessard** [comment]
@swift-ci smoke test and merge

**Comment by glessard** [comment]
@swift-ci please smoke test Linux platform

**Pull request merged**